compaction/compaction_manager.cc

```diff
@@ -863,7 +863,7 @@ auto swallow_enospc(const Ex& ex) noexcept {
 }
 
 void compaction_manager::do_stop() noexcept {
-    if (_state == state::none || _state == state::stopped) {
+    if (_state == state::none || _stop_future) {
```

Comment:
Would make stop_future a shared_future and waiting on it make it more robust? So we don't return before it's really stopped.

Reply:
Today, compaction_manager::stop() isn't reentrant. So we'd end up with a failure if we call it twice. The actual problem was the state changing during shutdown, and do_stop() (which is called both from abort source's subscription and stop()) performing the stop twice (note do_stop() doesn't wait, just trigger the action).